src/test/java/uk/ac/ebi/tsc/portal/api/team/controller/TeamRestControllerTest.java

```diff
@@ -902,6 +932,46 @@ public void testRemoveTeamContactTeamNotOwnerFail() throws AccountNotFoundExcept
 		subject.removeTeamContactEmail(request, principal, teamName, emailToRemove);
 	}
 
+	@Test
+	public void testAddToDefaultTeam() throws IOException {
+		resourceLoader = mock(ResourceLoader.class);
+		when(resourceLoader.getResource(defaultTeamsFile)).thenReturn(new InMemoryResource("[\n" +
+				"  {\n" +
+				"    \"emailDomain\":\"test.com\",\n" +
+				"    \"teamName\": \"TEST1\"\n" +
+				"  },\n" +
+				"  {\n" +
+				"    \"emailDomain\":\"test.org\",\n" +
+				"    \"teamName\": \"TEST2\"\n" +
+				"  },\n" +
+				"  {\n" +
+				"    \"emailDomain\":\"test.org\",\n" +
+				"    \"teamName\": \"TEST3\"\n" +
+				"  }\n" +
+				"]"));
+		teamService = new TeamService(teamRepository, accountService, domainService,
+				deploymentService, cppCopyService, depConfigService, applicationDeployer, sendMail, tokenService,
+				resourceLoader, ecpAapUsername, ecpAapPassword, defaultTeamsFile);
+		getPrincipal();
+		getAccount();
+		getRequest();
+		String teamName = "TEST1";
+		Team team = new Team();
+		team.setName(teamName);
+		String domainReference = "domainReference";
+		team.setDomainReference(domainReference);
+		Domain domain = mock(Domain.class);
+		when(domainService.getDomainByReference(anyString(), anyString())).thenReturn(domain);
+		when(domainService.addUserToDomain(any(), any(), any())).thenReturn(domain);
+		ReflectionTestUtils.setField(subject, "teamService", teamService);
+		when(tokenService.getAAPToken(ecpAapUsername, ecpAapPassword)).thenReturn(anyString());
+		when(teamRepository.findByName(teamName)).thenReturn(Optional.of(team));
+		when(teamRepository.findTeamByName(teamName)).thenReturn(Optional.of(team));
+		when(subject.addAccountToDefaultTeamsByEmail(request, response, principal)).thenCallRealMethod();
+		subject.addAccountToDefaultTeamsByEmail(request, response, principal);
+		assertTrue(team.accountsBelongingToTeam.size() == 1);
+	}
+
```

Comment:
Does not look like a unit test, unfortunately. Wrong place. The logic of adding to default team is happening in the service. The controller is just calling the method, so the unit test of the controller should just check that, preferably using a mock. The test of what the service is doing should be happening on the service level test.